### PR TITLE
Use EPOLL_CLOEXEC to prevent fd leaks.

### PR DESCRIPTION
**Pull request by cpuguy83**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:

/kind bug

**What this PR does / why we need it**:

Use EPOLL_CLOEXEC to prevent fd leaks.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

There's still an issue with fsnotify which is used in other places.
Reaching out to the maintainer, but he backed out of the project some
time ago and there's no active committers AFAIK.

**Does this PR introduce a user-facing change?**:

```release-note
Use O_CLOEXEC to ensure file descriptors do not leak to subprocesses.
```

### CODE REVIEW

**Review by dims — APPROVED** [review]

**Comment by timmycarr** [comment]
Hi! Friendly reminder from your release team: we are starting the code freeze for 1.15 tomorrow EOD. Seeing some good progress here, just checking in to see if this is still planned for the 1.15 cycle? See that milestone added a couple of days ago and awaiting a LGTM. Assuming that's the case. Can I get a quick confirmation here?

**Comment by soggiest** [comment]
@dims Is this release blocking? I see you've approved this PR but it is lacking a LGTM label, I'm inclined to move this to 1.16.

**Comment by dims** [comment]
@soggiest sounds good, my approval is not enough, need folks from sig-node to approve unfortunately :(

**Comment by derekwaynecarr** [comment]
apologies for the delay, for some reason this PR was lost in my review queue.  i assume you were observing leaks from this area, but its not clear at what rate?  we should have metrics for this and i am wondering why it hasn't come up prior.

@sjenning can you take a look?  i think you were the original author for using memcg notifications in this area, and it would be good for you to weigh in.

**Comment by sjenning** [comment]
related https://github.com/fsnotify/fsnotify/pull/219
/lgtm
I don't think we are leaking fd's that I know of but this is good hygiene unless you are intentionally wanting to share fd's with child processes (which we don't).

**Comment by cpuguy83** [comment]
Yes, I went on a hunt for possible leaks after a user reported some issues running out of inotify handles (without much to go on to debug with). These are just what I ran across as possible, and unlikely to want to be shared with sub-processes.

**Comment by dims** [comment]
@cpuguy83 let's rebase and ping folks for approval again :)

**Comment by derekwaynecarr** [comment]
thanks @cpuguy83 and @sjenning for commenting.

kubelet changes lgtm.

/approve
/lgtm

**Comment by dims** [comment]
/assign @thockin @dchen1107 

(We need pkg/ OWNERS approval, please see @derekwaynecarr 's `/approve` above)

**Comment by thockin** [comment]
Thanks!

/lgtm
/approve

**Pull request merged**
